lib/ipcs.c

```diff
@@ -78,6 +78,51 @@ qb_ipcs_create(const char *name,
 	return s;
 }
 
+qb_ipcs_service_t *
+qb_ipcs_alias_create(qb_ipcs_service_t *service,
+		     const char *name,
+		     enum qb_ipc_type type)
+{
+	struct qb_ipcs_service *s;
+
+	s = calloc(1, sizeof(struct qb_ipcs_service));
+	if (s == NULL) {
+		return NULL;
+	}
+	if (type == QB_IPC_NATIVE) {
+#ifdef DISABLE_IPC_SHM
+		s->type = QB_IPC_SOCKET;
+#else
+		s->type = QB_IPC_SHM;
+#endif /* DISABLE_IPC_SHM */
+	} else {
+		s->type = type;
+	}
+
+	s->pid = getpid();
+	s->needs_sock_for_poll = QB_FALSE;
+	s->poll_priority = QB_LOOP_MED;
+
+	/* Initial alloc ref */
+	qb_ipcs_ref(s);
+
+	s->service_id = service->service_id;
+	(void)strlcpy(s->name, name, NAME_MAX);
```

Comment:
To avoid duplication I'm thinking let qb_ipcs_create() take a NULL handlers, so this would just do that and set the handlers as below. Either that or functionize the common parts and have them both call that